src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/ValidateVariants.java

```diff
@@ -242,19 +261,29 @@ public void apply(final VariantContext vc, final ReadsContext readsContext, fina
 
             validateVariantsOrder(vc);
 
+            final boolean thisIntervalIsReference =  vc.getAlternateAlleles().size() == 1 && vc.getAlternateAllele(0).equals(Allele.NON_REF_ALLELE);
+
             // GenomeLocSortedSet will automatically merge intervals that are overlapping when setting `mergeIfIntervalOverlaps`
             // to true.  In a GVCF most blocks are adjacent to each other so they wouldn't normally get merged.  We check
             // if the current record is adjacent to the previous record and "overlap" them if they are so our set is as
             // small as possible while still containing the same bases.
+            if (previousInterval != null && previousInterval.overlapsWithMargin(refInterval, 0) &&
+                    (previousIntervalIsReference || thisIntervalIsReference)) {
```

Comment:
Is it not also illegal for a ref block to overlap with a variant?

Reply:
Yes, but this checks out right?  If previous is variant and this is ref, then throw; if previous is ref and this is variant then throw; if previous is ref and this is ref then throw; if previous is variant and this is variant then okay.  Anyway, I have tests for both now.

Reply:
AHHH I see || and && aren't the same thing... whoops....